grails-app/views/taglib/_displayLogo.gsp

```diff
@@ -1,8 +1,17 @@
+<%@ page import="org.pih.warehouse.core.RoleType" %>
+<%@ page import="org.pih.warehouse.core.User" %>
 <div class="logo">
     <g:if test="${attrs.includeLink}">
-        <a href="${createLink(uri: '/')}">
-            <img src="${attrs?.logoUrl}" />
-        </a>
+        <g:if test="${User.get(session?.user?.id)?.getHighestRole(session.warehouse).roleType == RoleType.ROLE_AUTHENTICATED}">
```

Comment:
One more point about the need for this lookup. I assume you were getting lazy init errors when trying to retrieve an association from session.user. I never should have set the full user object in the session. Instead we should store the user ID in the session and fetch the object on every request. We do this with the thread locals in AuthService so we could/should use that object whenever we need the current user or current location. Or we should hydrate the sesion.user and session.warehouse on every request similar to how we populate the thread locals. See SecurityFilters.